.github/workflows/ci.yml

```diff
@@ -17,272 +17,275 @@
-          sudo apt clean
-          docker rmi $(docker image ls -aq)
-          df -h
-      - name: "Tests [MySQL][Py3.6][integrations]"
+      - uses: actions/setup-python@v1
+        with:
+          python-version: '3.x'
+      - name: "Free space"
+        run: ./scripts/ci/ci_free_space_on_ci.sh
+      - name: "Build CI image ${{ matrix.python-version }}"
+        run: ./scripts/ci/ci_prepare_image_on_ci.sh
+      - name: "Tests"
         run: ./scripts/ci/ci_run_airflow_testing.sh
 
-  tests-p36-mysql-providers:
-    name: "Tests [MySQL5.7][Py3.7][providers][kerberos]"
-    runs-on: ubuntu-latest
-    needs: [statics, statics-tests]
-    env:
-      TRAVIS_JOB_NAME: "Tests [MySQL5.7][Py3.7][providers][kerberos]"
-      BACKEND: mysql
-      PYTHON_VERSION: 3.7
-      MYSQL_VERSION: 5.7
-      ENABLED_INTEGRATIONS: "kerberos"
-    steps:
-      - uses: actions/checkout@master
-      - name: "Tests [MySQL5.7][Py3.7][providers][kerberos]"
-        run: ./scripts/ci/ci_run_airflow_testing.sh tests/providers
-
-  tests-p37-mysql-kerberos:
-    name: "Tests [MySQL5.7][Py3.7][core][kerberos]"
-    runs-on: ubuntu-latest
-    needs: [statics, statics-tests]
-    env:
-      TRAVIS_JOB_NAME: "Tests [MySQL5.7][Py3.7][core][kerberos]"
-      BACKEND: mysql
-      PYTHON_VERSION: 3.7
-      MYSQL_VERSION: 5.7
-      ENABLED_INTEGRATIONS: "kerberos"
-    steps:
-      - uses: actions/checkout@master
-      - name: free disk space
-        run: |
-          sudo swapoff -a
-          sudo rm -f /swapfile
-          sudo apt clean
-          docker rmi $(docker image ls -aq)
-          df -h
-      - name: "Tests [MySQL5.7][Py3.7][core][kerberos]"
-        run: ./scripts/ci/ci_run_airflow_testing.sh --ignore=tests/providers
-
   requirements:
-    name: "Generate requirements"
+    name: "Requirements: Py ${{ matrix.python-version}} "
     runs-on: ubuntu-latest
-    needs: [statics, statics-tests]
     strategy:
       matrix:
         python-version: [3.6, 3.7]
+    env:
+      PYTHON_MAJOR_MINOR_VERSION: ${{ matrix.python-version }}
+      SHOW_GENERATE_REQUIREMENTS_INSTRUCTIONS: true
+      CI_JOB_TYPE: "Requirements"
     steps:
       - uses: actions/checkout@master
       - uses: actions/setup-python@v1
+      - name: "Build CI image ${{ matrix.python-version }}"
+        run: ./scripts/ci/ci_prepare_image_on_ci.sh
       - name: "Generate requirements"
-        env:
-          PYTHON_MAJOR_MINOR_VERSION: ${{ matrix.python-version }}
-          SHOW_GENERATE_REQUIREMENTS_INSTRUCTIONS: true
         run: ./scripts/ci/ci_generate_requirements.sh
 
-  build-image:
-    name: "Build production image"
+  push-images-to-github-cache:
+    name: "Build and push images to github registry"
     runs-on: ubuntu-latest
-    needs: [statics, statics-tests]
+    needs:
+      - tests-sqlite
+      - tests-postgres
+      - tests-mysql
+      - requirements
+      - prepare-backport-packages
+      - build-prod-image
+      - docs
```

Comment:
Nice one!

Reply:
Yes. unfortunately we cannot do one CI image build and store it in the registry. When you run pull requests from the forks, the GITHUB_TOKEN has only read permissions and you are not able to push image to Github Image Registry :(. And when you think about it, it makes perfect sense (Imagine someone changing the build and overriding all temporary images). However I believe (still need to verify it) that we can pull the images from the Github Registy rather than from the DockerHub and it should significantly speed up Pullng the images for rebuilds (it takes 3-4 minutes now). I yet has to confirm it.